src/bin/huak/commands/config/completion.rs

```diff
@@ -0,0 +1,63 @@
+use crate::commands::Cli;
+use crate::errors::CliResult;
+
+use huak::ops::config;
+
+use clap::{Args, Command, CommandFactory, Subcommand};
+use clap_complete::{generate, Shell};
+
+/// Prints the script to stdout and a way to add the script to the shell init file to stderr. This
+/// way if the user runs completion <shell> > completion.sh only the stdout will be redirected into
+/// completion.sh.
+pub fn run(config_command: Config) -> CliResult<()> {
+    match config_command.command {
+        ConfigCommand::Completion { shell } => {
+            generate_shell_completion_script(shell)
+        }
+        ConfigCommand::Install { shell } => {
+            let mut cmd: Command = Cli::command();
+            let _result = match shell {
+                Shell::Bash => config::_add_completion_bash(),
+                Shell::Elvish => config::_add_completion_elvish(),
+                Shell::Fish => config::_add_completion_fish(&mut cmd),
+                Shell::PowerShell => config::_add_completion_powershell(),
+                Shell::Zsh => config::_add_completion_zsh(&mut cmd),
+                _ => Ok(()),
+            };
+        }
+        ConfigCommand::Uninstall { shell } => {
+            let _result = match shell {
+                Shell::Bash => config::_remove_completion_bash(),
+                Shell::Elvish => config::_remove_completion_elvish(),
+                Shell::Fish => config::_remove_completion_fish(),
+                Shell::PowerShell => config::_remove_completion_powershell(),
+                Shell::Zsh => config::_remove_completion_zsh(),
+                _ => Ok(()),
+            };
+        }
+    }
+    Ok(())
+}
+
+fn generate_shell_completion_script(shell: Shell) {
+    let mut cmd = Cli::command();
+
+    generate(shell, &mut cmd, "huak", &mut std::io::stdout())
+}
+
+#[derive(Args)]
+pub struct Config {
+    #[command(subcommand)]
+    command: ConfigCommand,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum ConfigCommand {
+    /// Generates a shell completion script for supported shells.
+    /// See the help menu for more information on supported shells.
+    Completion { shell: Shell },
+    /// Installs the completion script in your shell init file.
+    Install { shell: Shell },
+    /// Uninstalls the completion script from your shell init file.
+    Uninstall { shell: Shell },
```

Comment:
Is the idea to use `huak config install` for setting up autocomplete? This feels wrong. Maybe just add `huak config completion --no-install` and `--uninstall`. Lmk what you think.

Reply:
Do you want the shell version to be behind a flag as well?

Reply:
You mean like `huak config completion <shell>` where shell is something like `bash`? If so nah I think it makes sense to just be`huak config completion <shell>` unless I'm missing something. I just think that using `huak config install` for *completion* configuration wouldn't make sense. If I'm a user I'm thinking "does this install configuation?" which seems ambiguous.